### PR TITLE
Add a SQLite job to the CI

### PR DESCRIPTION
**Pull request by elia**
While running specs locally I noticed that those two pending examples were successful.
Also restored another skipped commit.

## Checklist

Check out our [PR guidelines](https://github.com/solidusio/solidus/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [x] I have written a thorough PR description.
- [x] I have kept my commits small and atomic.
- [x] I have used clear, explanatory commit messages.

The following are not always needed (~cross them out~ if they are not):

- [x] I have added automated tests to cover my changes.
- [ ] ~~I have attached screenshots to demo visual changes.~~
- [ ] ~~I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).~~
- [ ] ~~I have updated the readme to account for my changes.~~

### CODE REVIEW

**Comment by waiting-for-dev** [comment]
Hey, @elia, thanks for your PR. Please, see https://github.com/solidusio/solidus/pull/4019#issuecomment-815473815

**Comment by elia** [comment]
thanks @waiting-for-dev good call, I'll remove the CI commit and keep the rest.

However it seems like the CI took the usual 13min, @kennyadsl can we resume #4019 now? 
Having local tests trigger this kind of PR from multiple people might be a hint we should have it covered.

**Comment by kennyadsl** [comment]
@elia it's taking way less now because Nebulab is taking care of the CI costs (we should probably pay with Open Collective funds for that, but that's another story) and we have more jobs in parallel. That said, it may be time to resume #4019 now, or incorporate that same change in this PR.

**Comment by elia** [comment]
@kennyadsl thanks! I'll add that commit to this PR and open for review once green 👍

File: .circleci/config.yml [hunk 1]
```diff
@@ -37,6 +37,14 @@ executors:
...
       - image: cimg/mysql:5.7
 
+  sqlite:
+    working_directory: *workdir
+    environment:
+      <<: *environment
+      DB: sqlite
+    docker:
+      - image: *image
+
```
Comment (forkata) [inline .circleci/config.yml:47]: Do you think we should add this to the https://github.com/solidusio/circleci-orbs-extensions repo as well so extensions also can opt-in to running specs against SQLite on CI?

**Review by forkata — APPROVED** [review]
🎉 Thanks @elia, these changes look great!

File: api/spec/requests/spree/api/checkouts_spec.rb [hunk 2]
```diff
@@ -82,8 +82,6 @@ module Spree::Api
         end
 
         it "can update addresses and transition from address to delivery" do
-          pending "SQLite adapter fails to provide a unique index" if ActiveRecord::Base.connection.adapter_name == "SQLite"
```
Comment (waiting-for-dev) [inline api/spec/requests/spree/api/checkouts_spec.rb:85]: I confirmed with the Docker setup that the tests are green now :four_leaf_clover:

File: core/lib/spree/testing_support/dummy_app.rb [hunk 3]
```diff
@@ -38,7 +38,7 @@ def self.setup(gem_root:, lib_name:, auto_migrate: true)
     ENV["LIB_NAME"] = lib_name
     DummyApp::Application.config.root = File.join(gem_root, 'spec', 'dummy')
 
-    DummyApp::Application.initialize!
+    DummyApp::Application.initialize! unless DummyApp::Application.initialized?
```
Comment (waiting-for-dev) [inline core/lib/spree/testing_support/dummy_app.rb:41]: :heart:

File: core/lib/spree/testing_support/dummy_app.rb [hunk 4]
```diff
@@ -116,6 +116,7 @@ class Application < ::Rails::Application
     end
     config.paths['db/migrate'] = migration_dirs
     ActiveRecord::Migrator.migrations_paths = migration_dirs
+    ActiveRecord::Migration.verbose = false
```
Comment (waiting-for-dev) [inline core/lib/spree/testing_support/dummy_app.rb:119]: :heart:

**Review by waiting-for-dev — APPROVED** [review]
Thanks, @elia!

**Review by kennyadsl — APPROVED** [review]

**Comment by elia** [comment]
@forkata sure! I'll work that today 👍
I'd also feel more confident if we open the test matrix to all supported ruby and rails versions, both in solidus and for extensions, but let's take one step at a time 😄

File: .circleci/config.yml (hunk 1 above)
Comment (elia) [reply .circleci/config.yml:47]: Added in https://github.com/solidusio/circleci-orbs-extensions/pull/48, will be released in v0.4.0

**Pull request merged**